odl/tomo/geometry/parallel.py

```diff
@@ -19,76 +19,122 @@
-        det_init_pos : `array-like`
-            Initial position of the detector reference point
+            The detector to use in this geometry.
+        det_pos_init : `array-like`
+            Initial position of the detector reference point.
+        kwargs :
+            Additional parameters passed on to the ``__init__`` method
+            of `Geometry`.
         """
-        super().__init__(ndim, apart, detector)
+        super().__init__(ndim, apart, detector, **kwargs)
 
         if self.ndim not in (2, 3):
-            raise ValueError('number of dimensions is {}, expected 2 or 3'
-                             ''.format(ndim))
+            raise ValueError('`ndim` must be 2 or 3, got {}'.format(ndim))
+
+        self.__det_pos_init = np.asarray(det_pos_init, dtype='float64')
+        if self.det_pos_init.shape != (self.ndim,):
+            raise ValueError('`det_pos_init` must have shape ({},), got {}'
+                             ''.format(self.ndim, self.det_pos_init.shape))
 
-        self._det_init_pos = np.asarray(det_init_pos, dtype='float64')
-        if self._det_init_pos.shape != (self.ndim,):
-            raise ValueError('initial detector position has shape {}, '
-                             'expected ({},)'
-                             ''.format(self._det_init_pos.shape, self.ndim))
+    @property
+    def det_pos_init(self):
+        """Initial position of the detector reference point."""
+        return self.__det_pos_init
 
     @property
     def angles(self):
         """Discrete angles given in this geometry."""
         return self.motion_grid.coord_vectors[0]
 
-    def det_refpoint(self, angles):
+    def det_refpoint(self, angle):
         """Return the position of the detector ref. point at ``angles``.
 
         The reference point is given by a rotation of the initial
         position by ``angles``.
 
+        For an angle ``phi``, the detector position is given by ::
+
+            det_ref(phi) = translation +
+                           rot_matrix(phi) * (det_pos_init - translation)
+
+        where ``det_pos_init`` is the detector reference point at initial
+        state.
+
         Parameters
         ----------
-        angles : float
-            Parameters describing the detector rotation, must be
+        angle : float
+            Parameter describing the detector rotation, must be
             contained in `motion_params`.
 
         Returns
         -------
         point : `numpy.ndarray`, shape (`ndim`,)
-            The reference point for the given parameters
+            The reference point for the given parameter.
+
+        Examples
+        --------
+        For 2d and default arguments, the detector starts at ``e_y`` and
+        rotates to ``-e_x`` at 90 degrees:
+
+        >>> apart = odl.uniform_partition(0, np.pi, 10)
+        >>> dpart = odl.uniform_partition(-1, 1, 20)
+        >>> geom = Parallel2dGeometry(apart, dpart)
+        >>> np.allclose(geom.det_refpoint(0), [0, 1])
```

Comment:
If at all possible, try to make this mroe human readable (i.e. make this actually return [0, 1] or something. These `allclose` in doctests is weird and deviates somewhat from their examples idea.